test/DotNetWorker.OpenTelemetry.Tests/EndToEndTests.cs

```diff
@@ -0,0 +1,164 @@
+        Assert.Equal("azure_functions", resource.Attributes.FirstOrDefault(a => a.Key == "cloud.platform").Value);
+    }
+
+    [Fact]
+    public void ResourceDetectorLocalDevelopment2()
+    {
+        using var _ = SetupDefaultEnvironmentVariables();
+
+        FunctionsResourceDetector detector = new FunctionsResourceDetector();
+        Resource resource = detector.Detect();
+
+        Assert.Equal($"/subscriptions/AAAAA-AAAAA-AAAAA-AAA/resourceGroups/rg/providers/Microsoft.Web/sites/appName"
+            , resource.Attributes.FirstOrDefault(a => a.Key == "cloud.resource.id").Value);
+        Assert.Equal($"EastUS", resource.Attributes.FirstOrDefault(a => a.Key == "cloud.region").Value);
+    }
+
+    private static IDisposable SetupDefaultEnvironmentVariables()
+    {
+        return new TestScopedEnvironmentVariable(new Dictionary<string, string>
+        {
+            { "WEBSITE_SITE_NAME", "appName" },
+            { "WEBSITE_RESOURCE_GROUP", "rg" },
+            { "WEBSITE_OWNER_NAME", "AAAAA-AAAAA-AAAAA-AAA+appName-EastUSwebspace" },
+            { "REGION_NAME", "EastUS" }
+        });
+    }
+
+    internal class OtelFunctionDefinition : FunctionDefinition
+    {
+        public static readonly string DefaultPathToAssembly = typeof(OtelFunctionDefinition).Assembly.Location;
+        public static readonly string DefaultEntryPoint = $"{typeof(OtelFunctionDefinition).FullName}.{nameof(TestFunction)}";
+        public static readonly string DefaultId = "TestId";
+        public static readonly string DefaultName = "TestName";
+
+        public OtelFunctionDefinition()
+        {
+            Parameters = (new[] { new FunctionParameter("context", typeof(FunctionContext)) }).ToImmutableArray();
+        }
+
+        public override ImmutableArray<FunctionParameter> Parameters { get; }
+
+        public override string PathToAssembly { get; } = DefaultPathToAssembly;
+
+        public override string EntryPoint { get; } = DefaultEntryPoint;
+
+        public override string Id { get; } = DefaultId;
+
+        public override string Name { get; } = DefaultName;
+
+        public override IImmutableDictionary<string, BindingMetadata> InputBindings { get; } = ImmutableDictionary<string, BindingMetadata>.Empty;
+
+        public override IImmutableDictionary<string, BindingMetadata> OutputBindings { get; } = ImmutableDictionary<string, BindingMetadata>.Empty;
+
+        public static Activity LastActivity;
+
+        public void TestFunction(FunctionContext context)
+        {
+            LastActivity = Activity.Current;
+            Activity.Current.ActivityTraceFlags = ActivityTraceFlags.Recorded;
+            Activity.Current.AddTag("CustomKey", "CustomValue");
+
+            var logger = context.GetLogger("TestFunction");
+            logger.LogWarning("Test");
+
+            HttpClient httpClient = new HttpClient();
+            httpClient.GetAsync("https://www.bing.com").GetAwaiter().GetResult();
```

Comment:
Instead of making an actual call, I would create a custom `HttpMessageHandler` where `SendAsync` is mocked to return the `HttpResponseMessage` you want. You then pass this to the ctor of `HttpClient`.

Reply:
sure, I am also trying to use InMemoryExporter to validate telemetry. I will improve this UT later.